src/rdkv/impl.c

```diff
@@ -340,33 +335,117 @@ void populatePersistenceData()
+            if(!pTeleSuplUrl)
+            {
+                WebcfgInfo("Supl_url empty in db json file. Set default value\n");
+                snprintf(webCfgPersist.m_teleSuplUrl, 1024, "%s", DEF_SUPL_URL);
+            }
+            else
+            {
+                WebcfgInfo("Sup_url present in db json file\n");
+                snprintf(webCfgPersist.m_teleSuplUrl, 1024, "%s", cJSON_GetStringValue(pTeleSuplUrl));
+            }
+
+            cJSON *pRfc = cJSON_GetObjectItem(pParser, WEBCFG_RFC_PARAM);
+            if(!pRfc)
+            {
+                WebcfgInfo("Rfc empty in db json file. Set default value\n");
+                snprintf(webCfgPersist.m_rfcStatus, 16, "%s", "true");
+            }
+            else
+            {
+                WebcfgInfo("Rfc present in db json file\n");
+                snprintf(webCfgPersist.m_rfcStatus, 16, "%s", cJSON_GetStringValue(pRfc));
+            }
+        }
     }
     else
     {
-        char *pPartnerId = NULL;
-        pPartnerId = getPartnerID();
-        cJSON *pParser = load_partner_json_file(WEBCFG_PARTNER_JSON_FILE);
-        cJSON *pItem = cJSON_GetObjectItem(pParser, pPartnerId);
+        WebcfgInfo("DB file is not present. Regenerate file & contents\n");
+        load_partnerid_params();
+    }
+}
+
+
+void load_partnerid_params()
+{
+    char *pPartnerId = NULL;
+    pPartnerId = getPartnerID();
+    if(!pPartnerId)
+    {
+        pPartnerId="default";
+        WebcfgInfo("Partner id is default.\n");
+    }
+    cJSON *pParser = load_partner_json_file(WEBCFG_PARTNER_JSON_FILE);
+    if(!pParser)
+    {
+        WebcfgInfo("Parsing from main json file is empty. Load default values\n");
+        load_def_values();
+        return;
+    }
+    cJSON *pItem = cJSON_GetObjectItem(pParser, pPartnerId);
+    if(pItem)
+    {
         cJSON *name = cJSON_CreateString("true");
         cJSON_AddItemToObject(pItem, WEBCFG_RFC_PARAM, name);
         char *pValue2 = cJSON_Print(pItem);
-
-
+        if((pValue2) && (strlen(pValue2)))
         {
+            WebcfgInfo("Parsing item/get item from parser\n");
             cJSON *pUrl = cJSON_GetObjectItemCaseSensitive(pItem, WEBCFG_URL_PARAM);
             cJSON *pTeleSuplUrl = cJSON_GetObjectItemCaseSensitive(pItem, WEBCFG_SUPPLEMENTARY_TELEMETRY_PARAM);
 
             snprintf(webCfgPersist.m_rfcStatus, 16, "%s", "true");
             snprintf(webCfgPersist.m_url, 1024, "%s", cJSON_GetStringValue(pUrl));
             snprintf(webCfgPersist.m_teleSuplUrl, 1024, "%s", cJSON_GetStringValue(pTeleSuplUrl));
+
+            WebcfgInfo("Writing from json obj to db file.\n");
+            writeToFile(pValue2);
         }
+    }
+    else
+    {
+        WebcfgInfo("pItem/get item from parser is empty. Load default values\n");
+        load_def_values();
+        return;
+    }
+
+}
+
+void load_def_values()
+{
+    WebcfgInfo("Load default values to db json\n");
```

Comment:
Can this be changed to call existing function?? Ref: Line#540

ie,
```
void load_def_values()
{
     snprintf(webCfgPersist.m_rfcStatus, 16, "%s", "true");
     snprintf(webCfgPersist.m_url, 1024, "%s", DEF_WEB_URL);
     snprintf(webCfgPersist.m_teleSuplUrl, 1024, "%s", DEF_SUPL_URL);
     cJSON * pWebCfg = convertWebCfgDataToJson();
     if (pWebCfg)
     {
            char* pString = cJSON_Print(pWebCfg);
            writeToFile(pString);
            cJSON_Delete(pWebCfg);
     }
}
```

Reply:
I have made the respective changes in load_def_values()